### PR TITLE
Grafana/UI: Add basic legend to the PieChart

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What this PR does / why we need it**:
This adds a basic legend to the PieChart core plugin

### CODE REVIEW

File: packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+import { VizLegend } from '../VizLegend/VizLegend';
+import { PieChart, DEFAULT_COLOR, Props as PieChartProps } from './PieChart';
 
 export interface Props extends PieChartProps {}
 
-export const PieChartWithLegend: FC<Props> = ({ width, height, ...restProps }) => {
-  return <div>Need VizLayout in grafana/ui</div>;
+export const PieChartWithLegend: FC<Props> = ({ values, width, height, ...restProps }) => {
+  const legendItems = values.map<VizLegendItem>((value) => {
+    console.log(value.title, value.color);
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx:11]: This should probably be removed.

Comment (oscarkilhed) [reply packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx:11]: Oh, dammit. Even greped the diff for console logs to get rid of them all, but forgot to save this file after the change. Thanks.

File: packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx [hunk 2]
```diff
@@ -1,8 +1,29 @@
...
+  const legendItems = values.map<VizLegendItem>((value) => {
+    return {
+      label: value.title ?? '',
+      color: value.color ?? DEFAULT_COLOR,
+      yAxis: 1,
+    };
+  });
+
+  const legendElement = (
+    <VizLegend items={legendItems} placement="right" displayMode={LegendDisplayMode.List}></VizLegend>
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx:19]: Maybe change to this:
```suggestion
    <VizLegend items={legendItems} placement="right" displayMode={LegendDisplayMode.List} />
```

File: packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
+  );
+
+  return (
+    <VizLayout width={width} height={height} legend={legendElement}>
+      {(vizWidth: number, vizHeight: number) => (
+        <PieChart values={values} width={vizWidth} height={vizHeight} {...restProps}></PieChart>
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx:25]: Maybe change to this:
```suggestion
        <PieChart values={values} width={vizWidth} height={vizHeight} {...restProps} />
```

**Review by hugohaggmark — COMMENTED** [review]
Looking Good, I think the ComponentWithLegend approach is something we tested with the old Graph and I think most of us agree it wasn't a very good pattern. 
Not sure this is the scope for this PR though, but you might want to look at how we compose Legend and other abilities in the new Graph as an inspiration moving forward.

Also a friendly reminder to always add Project and Milestone on your PRs. And if this is going into the v.7.4.x branch it will also need the `backport v7.4.x` label.

**Review by torkelo — APPROVED** [review]
Looks good!  

Think instead of exporting PieChartWithĹegend, lets make the exported PieChart component support legend. Maybe rename PieChart to PieChartSvg or something.  

The panel needs some legend options. Like placement and what values to show. But can be done in a follow-up PR

File: packages/grafana-ui/src/components/PieChart/PieChart.tsx [hunk 4]
```diff
@@ -10,6 +10,8 @@ import { useTooltip, useTooltipInPortal } from '@visx/tooltip';
 import { useComponentInstanceId } from '../../utils/useComponetInstanceId';
 import { css } from 'emotion';
 
+export const DEFAULT_COLOR = 'gray';
```
Comment (torkelo) [inline packages/grafana-ui/src/components/PieChart/PieChart.tsx:13]: think there is a FALLBACK_COLOR exported from grafana/data

**Comment by oscarkilhed** [comment]
> Think instead of exporting PieChartWithĹegend, lets make the exported PieChart component support legend. Maybe rename PieChart to PieChartSvg or something.

Ah, missed this. I removed the PieChartWithLegend and added the legend to the PieChart. But you're probably right in that the svg should be separated from the legend

**Review by torkelo — APPROVED** [review]

**Pull request merged**